x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,327 @@
+                ...serializeTimeRange(),
+                ...serializeFieldStatsChartState(),
+              },
+              references,
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...fieldStatsControlsComparators,
+        }
+      );
+
+      const reload$ = fetch$(api).pipe(
+        skipWhile((fetchContext) => !fetchContext.isReload),
+        map(() => Date.now())
+      );
+      const query$ = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.query),
+        distinctUntilChanged(fastIsEqual)
+      );
+      const filters$ = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.filters),
+        distinctUntilChanged(fastIsEqual)
+      );
+      const reset$ = resetData$.pipe(skip(1), distinctUntilChanged());
+
+      const appliedTimeRange$: Observable<TimeRange | undefined> = fetch$(api).pipe(
+        map((fetchContext) => fetchContext.timeRange),
+        distinctUntilChanged(fastIsEqual)
+      );
+
+      const onTableUpdate = (changes: Partial<DataVisualizerTableState>) => {
+        if (isDefined(changes?.showDistributions)) {
+          fieldStatsControlsApi.showDistributions$.next(changes.showDistributions);
+        }
+      };
+      const statsTableCss = css({
+        width: '100%',
+        height: '100%',
+        overflowY: 'auto',
+      });
+
+      return {
+        api,
+        Component: () => {
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
+          }
+
+          const [dataViews, esqlQuery, viewType, showPreviewByDefault] =
+            useBatchedPublishingSubjects(
+              api.dataViews,
+              api.query$,
+              api.viewType$,
+              api.showDistributions$
+            );
+
+          const timeRange = useObservable(appliedTimeRange$, undefined);
+          const globalQuery = useObservable(query$, undefined);
+          const globalFilters = useObservable(filters$, undefined);
+          const lastReloadRequestTime = useObservable(reload$, Date.now());
+
+          const isEsqlMode = viewType === FieldStatsInitializerViewType.ESQL;
+
+          const dataView =
+            Array.isArray(dataViews) && dataViews.length > 0 ? dataViews[0] : undefined;
+          const onAddFilter = (
+            field: string | DataViewField,
+            value: string,
+            operator: '+' | '-'
+          ) => {
+            if (!dataView || !pluginStart.data) return;
+            let filters = generateFilters(
+              pluginStart.data.query.filterManager,
+              field,
+              value,
+              operator,
+              dataView
+            );
+            filters = filters.map((filter) => ({
+              ...filter,
+              $state: { store: FilterStateStore.APP_STATE },
+            }));
+            pluginStart.data.query.filterManager.addFilters(filters);
```

Comment:
Instead of relying on this singleton. Could you instead execute the action `ACTION_GLOBAL_APPLY_FILTER`. Here is an example you can follow for executing this action https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/react_embeddable/initialize_action_handlers.ts#L31

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/f990d2278cc309c21b0893cd076b8e2f2ec127b1